### DISCUSSION TITLE
Issue on C++/WinRT app consumes a C# component with WinUI3 controls

### DISCUSSION THREAD

**Discussion by lcmsft** [microsoft/CsWinRT#1199] [Q&A] [answered]
I follow this article to consume c# UI component in C++/WinRT app. 
https://docs.microsoft.com/en-us/windows/apps/develop/platform/csharp-winrt/create-winrt-component-winui-cswinrt

But I got an error saying "Payload contains two or more files with the same destination path "WinRT.Runtime.dll"

![image](https://user-images.githubusercontent.com/91639102/171527625-131f9ff7-2f89-47f9-ab49-223c4006b92a.png)

I looked at the samples here https://github.com/microsoft/CsWinRT/tree/master/src/Samples/AuthoringDemo

But unfortunately, the WinUI3CppApp is not consuming a UI component. 

Could anyone share a working sample with me that a C# UI component is consumed by C++ App? 

Thanks!
Lei

email: lecao@microsoft.com

**Comment by angelazhangmsft** [comment, marked as answer]
@lcmsft this looks similar to the issue https://github.com/microsoft/CsWinRT/issues/1151 and the current workaround is to set `PublishReadyToRun` to `False`. That issue should be resolved for 1.1 GA.

Here is a working sample: https://github.com/microsoft/WindowsAppSDK-Samples/tree/main/Samples/CustomControls

cc @j0shuams

**Comment by lcmsft** [comment]
Thanks. The sample provided works!
